internal/handlers/pg/pgdb/collections.go

```diff
@@ -187,6 +189,22 @@ func DropCollection(ctx context.Context, tx pgx.Tx, db, collection string) error
 	return nil
 }
 
+// RenameCollection changes the name of an existing collection.
+//
+// It returns ErrTableNotExist if either source database or collection does not exist.
+// It returns ErrAlreadyExist if the target database or collection already exists.
+// It returns ErrInvalidCollectionName if collection name is not valid.
+func RenameCollection(ctx context.Context, tx pgx.Tx, db, from, to string) error {
+	if !validateCollectionNameRe.MatchString(to) ||
+		strings.HasPrefix(to, reservedPrefix) ||
+		!utf8.ValidString(to) ||
+		len(to) > maxTableNameLength {
```

Comment:
See https://github.com/FerretDB/FerretDB/pull/2343/files#r1179874503

We should do this check of `len(to) > maxTableNameLength` not here. But before we call `RenameCollection` :hugs: